paddle/fluid/framework/ir/mkldnn/mkldnn_placement_pass_tester.cc

```diff
@@ -21,112 +21,129 @@ namespace paddle {
+  // operator                      use_mkldnn
+  // ---------------------------------------
+  // (a,b)->concat->c              none
+  // (c,weights,bias)->conv->f     none
+  // f->relu->g                    false
+  // g->pool->h                    false
+  // (h,weights2,bias2)->conv->k   true
+  // k->relu->l                    true
+  ProgramDesc BuildProgramDesc() {
+    ProgramDesc prog;
+
+    for (auto& v :
+         std::vector<std::string>({"a", "b", "c", "weights", "bias", "f", "g",
+                                   "h", "weights2", "bias2", "k", "l"})) {
+      auto* var = prog.MutableBlock(0)->Var(v);
+      var->SetType(proto::VarType::SELECTED_ROWS);
+      if (v == "weights" || v == "bias") {
+        var->SetPersistable(true);
+      }
     }
+
+    SetOp(&prog, "concat", "concat1", std::vector<std::string>({"a", "b"}),
+          std::vector<std::string>({"c"}), boost::indeterminate);
+    SetOp(&prog, "conv2d", "conv1",
+          std::vector<std::string>({"c", "weights", "bias"}),
+          std::vector<std::string>({"f"}), boost::indeterminate);
+    SetOp(&prog, "relu", "relu1", std::vector<std::string>({"f"}),
+          std::vector<std::string>({"g"}), false);
+    SetOp(&prog, "pool2d", "pool1", std::vector<std::string>({"g"}),
+          std::vector<std::string>({"h"}), false);
+    SetOp(&prog, "conv2d", "conv2",
+          std::vector<std::string>({"h", "weights2", "bias2"}),
+          std::vector<std::string>({"k"}), true);
+    SetOp(&prog, "relu", "relu2", std::vector<std::string>({"k"}),
+          std::vector<std::string>({"l"}), true);
+
+    return prog;
   }
 
-  SetOp(&prog, "concat", "concat1", std::vector<std::string>({"a", "b"}),
-        std::vector<std::string>({"c"}), boost::indeterminate);
-  SetOp(&prog, "conv2d", "conv1",
-        std::vector<std::string>({"c", "weights", "bias"}),
-        std::vector<std::string>({"f"}), boost::indeterminate);
-  SetOp(&prog, "relu", "relu1", std::vector<std::string>({"f"}),
-        std::vector<std::string>({"g"}), false);
-  SetOp(&prog, "pool2d", "pool1", std::vector<std::string>({"g"}),
-        std::vector<std::string>({"h"}), false);
-  SetOp(&prog, "conv2d", "conv2",
-        std::vector<std::string>({"h", "weights2", "bias2"}),
-        std::vector<std::string>({"k"}), true);
-  SetOp(&prog, "relu", "relu2", std::vector<std::string>({"k"}),
-        std::vector<std::string>({"l"}), true);
-
-  return prog;
-}
+ public:
+  void MainTest(std::initializer_list<std::string> mkldnn_enabled_op_types,
+                unsigned expected_use_mkldnn_true_count) {
+    auto prog = BuildProgramDesc();
 
-void MainTest(std::initializer_list<std::string> mkldnn_enabled_op_types,
-              unsigned expected_use_mkldnn_true_count) {
-  auto prog = BuildProgramDesc();
+    std::unique_ptr<ir::Graph> graph(new ir::Graph(prog));
 
-  std::unique_ptr<ir::Graph> graph(new ir::Graph(prog));
+    auto pass = PassRegistry::Instance().Get("mkldnn_placement_pass");
```

Comment:
LGTM.   I just have one doubt,fter we get this mkldnn pass, do we always expect its GetPlaceName() is "MKLDNN" ?

Reply:
Yes. The `MKLDNNPlacementPass` and `CUDNNPlacementPass` classes override the `PlacementPassBasebase` class' method `GetPlacementName()`. They return `MKLDNN` and `cuDNN`, respectively.